alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -379,282 +351,114 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
-
-            alertContexts.add(
-                AlertContext(
-                    alert = alert,
-                    associatedQueries = alert.findingIds.flatMap { findingId ->
-                        monitorCtx.findingsToTriggeredQueries?.getOrDefault(findingId, emptyList()) ?: emptyList()
-                    },
-                    sampleDocs = listOfNotNull(docSource)
-                )
-            )
-        }
-
-        val shouldDefaultToPerExecution = defaultToPerExecutionAction(
-            monitorCtx.maxActionableAlertCount,
-            monitorId = monitor.id,
-            triggerId = trigger.id,
-            totalActionableAlertCount = alerts.size,
-            monitorOrTriggerError = actionCtx.error
-        )
-
-        for (action in trigger.actions) {
-            val actionExecutionScope = action.getActionExecutionPolicy(monitor)!!.actionExecutionScope
-            if (actionExecutionScope is PerAlertActionScope && !shouldDefaultToPerExecution) {
-                for (alertContext in alertContexts) {
-                    val actionResults = this.runAction(action, actionCtx.copy(alerts = listOf(alertContext)), monitorCtx, monitor, dryrun)
-                    triggerResult.actionResultsMap.getOrPut(alertContext.alert.id) { mutableMapOf() }
-                    triggerResult.actionResultsMap[alertContext.alert.id]?.set(action.id, actionResults)
-                }
-            } else if (alertContexts.isNotEmpty()) {
-                val actionResults = this.runAction(action, actionCtx.copy(alerts = alertContexts), monitorCtx, monitor, dryrun)
-                for (alert in alerts) {
-                    triggerResult.actionResultsMap.getOrPut(alert.id) { mutableMapOf() }
-                    triggerResult.actionResultsMap[alert.id]?.set(action.id, actionResults)
-                }
-            }
-        }
+    private fun buildTriggerResults(
+        docLevelMonitorFanOutResponses: MutableList<DocLevelMonitorFanOutResponse>,
+    ): MutableMap<String, DocumentLevelTriggerRunResult> {
+        val triggerResults = mutableMapOf<String, DocumentLevelTriggerRunResult>()
+        val triggerErrorMap = mutableMapOf<String, MutableList<AlertingException>>()
+        for (res in docLevelMonitorFanOutResponses) {
+            for (triggerId in res.triggerResults.keys) {
+                val documentLevelTriggerRunResult = res.triggerResults[triggerId]
+                if (documentLevelTriggerRunResult != null) {
+                    if (false == triggerResults.contains(triggerId)) {
+                        triggerResults[triggerId] = documentLevelTriggerRunResult
+                        triggerErrorMap[triggerId] = if (documentLevelTriggerRunResult.error != null) {
+                            val error = if (documentLevelTriggerRunResult.error is AlertingException) {
+                                documentLevelTriggerRunResult.error as AlertingException
```

Comment:
Style nit: 6x nested. `buildTriggerResults` as a whole should be broken in a few private methods to improve readability